packages/core/useDevicePlatform/index.ts

```diff
@@ -0,0 +1,97 @@
+import { readonly, ref, watch } from 'vue'
+import { type ConfigurableWindow, defaultWindow } from '../_configurable'
+
+type OperatingSystem =
+  | 'iOS'
+  | 'Android OS'
+  | 'BlackBerry OS'
+  | 'Windows Mobile'
+  | 'Amazon OS'
+  | 'Windows 3.11'
+  | 'Windows 95'
+  | 'Windows 98'
+  | 'Windows 2000'
+  | 'Windows XP'
+  | 'Windows Server 2003'
+  | 'Windows Vista'
+  | 'Windows 7'
+  | 'Windows 8'
+  | 'Windows 8.1'
+  | 'Windows 10'
+  | 'Windows ME'
+  | 'Windows CE'
+  | 'Windows 11'
+  | 'Open BSD'
+  | 'Sun OS'
+  | 'Linux'
+  | 'Mac OS'
+  | 'QNX'
+  | 'BeOS'
+  | 'OS/2'
+  | 'Chrome OS'
+
+type OperatingSystemRule = [OperatingSystem, RegExp]
+
+const operatingSystemRules: OperatingSystemRule[] = [
+  ['iOS', /iP(hone|od|ad)/],
+  ['Android OS', /Android/],
+  ['BlackBerry OS', /BlackBerry|BB10/],
+  ['Windows Mobile', /IEMobile/],
+  ['Amazon OS', /Kindle/],
+  ['Windows 3.11', /Win16/],
+  ['Windows 95', /(Windows 95)|(Win95)|(Windows_95)/],
+  ['Windows 98', /(Windows 98)|(Win98)/],
+  ['Windows 2000', /(Windows NT 5.0)|(Windows 2000)/],
+  ['Windows XP', /(Windows NT 5.1)|(Windows XP)/],
+  ['Windows Server 2003', /(Windows NT 5.2)/],
+  ['Windows Vista', /(Windows NT 6.0)/],
+  ['Windows 7', /(Windows NT 6.1)/],
+  ['Windows 8', /(Windows NT 6.2)/],
+  ['Windows 8.1', /(Windows NT 6.3)/],
+  ['Windows 10', /(Windows NT 10.0)/],
+  ['Windows 11', /(Windows NT 11.0)/],
+  ['Windows ME', /Windows ME/],
+  ['Windows CE', /Windows CE|WinCE|Microsoft Pocket Internet Explorer/],
+  ['Open BSD', /OpenBSD/],
+  ['Sun OS', /SunOS/],
+  ['Chrome OS', /CrOS/],
+  ['Linux', /(Linux)|(X11)/],
+  ['Mac OS', /(Mac_PowerPC)|(Macintosh)/],
+  ['QNX', /QNX/],
+  ['BeOS', /BeOS/],
+  ['OS/2', /OS\/2/],
+]
+
+function _detectOS(ua: string): OperatingSystem | null {
+  for (let i = 0, count = operatingSystemRules.length; i < count; i++) {
+    const [os, regex] = operatingSystemRules[i]
+    const match = regex.exec(ua)
+
+    if (match)
+      return os
+  }
+
+  return null
+}
+
+export function useDevicePlatform(options: ConfigurableWindow = {}) {
+  const { window = defaultWindow } = options
+
+  const os = ref<string | null>(null)
```

Comment:
cannot use here `OperatingSystem` instead string?

if so, we'll need to also export `OperatingSystem` (to avoid problems with Typescript >= 5.6 about required anotation)